src/modules/settings/index.js

```diff
@@ -228,40 +54,7 @@ class SettingsModule {
 
   openSettings(e) {
     e.preventDefault();
-    $('#bttvSettingsPanel').show('slow');
-  }
-
-  updateSettingToggle(settingId, value) {
-    $(`#${settingId}True`).prop('checked', value === true);
-    $(`#${settingId}False`).prop('checked', value === false);
-  }
-
-  backup() {
-    const rv = storage.getStorage();
-    save(JSON.stringify(rv), 'bttv_settings.backup');
-  }
-
-  import(target) {
-    getDataURLFromUpload(target, (data) => {
-      if (!isJSON(data)) return;
-
-      const settingsToImport = JSON.parse(data);
-      Object.keys(settingsToImport).forEach((s) => settings.set(s.split('bttv_')[1], settingsToImport[s]));
-
-      setTimeout(() => window.location.reload(), 1000);
-    });
-  }
-
-  doSearch() {
-    const val = $('#bttvSettingsSearch').val().trim().toLowerCase();
-    if (val === '') {
-      $('[class^="option bttvOption-"]').css('display', 'block');
-      return;
-    }
-    settings.getSettings().forEach((setting) => {
-      const shouldShow = setting.name.toLowerCase().includes(val) || setting.description.toLowerCase().includes(val);
-      $(`.bttvOption-${setting.id}`).css('display', shouldShow ? 'block' : 'none');
-    });
+    ReactDOM.render(<BTTVWindow open />, document.getElementById('bttvSettingsPanel'));
```

Comment:
rather than rerendering, have you considered passing in a callback prop which sets the state of the modal when called?